driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -280,6 +282,7 @@ public class ConnectionString {
     private Integer maxConnectionLifeTime;
     private Integer maxConnecting;
     private Integer connectTimeout;
+    private Long timeout;
```

Comment:
should this have the unit, `timeoutMS`?

Reply:
Added to the javadocs its milliseconds. Its annoying that its inconsistent but I opted to keep it consistently inconsistent as not all timeouts will be deprecated.